source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -144,11 +144,7 @@ Filter::StreamOpenState Filter::openStream() {
 
 void Filter::closeStream() {
   if (stream_) {
-    if (grpc_service_.has_envoy_grpc()) {
-      logging_info_->setBytesSent(stream_->streamInfo().bytesSent());
-      logging_info_->setBytesReceived(stream_->streamInfo().bytesReceived());
-      logging_info_->setUpstreamHost(stream_->streamInfo().upstreamInfo()->upstreamHost());
-    }
+    logGrpcStreamInfo();
```

Comment:
The logging in close stream is still needed for remoteClose cases like error, localReply. For example [here](https://github.com/envoyproxy/envoy/blob/603aa03a9534ae258395bf6942200f52bfd4df75/source/extensions/filters/http/ext_proc/ext_proc.cc#L714) the stream is closed before `onFinishProcessorCall`

We probably can try to unify them but I think there is an edge case that close side stream without hitting ``onFinishProcessorCall` at all , which is shown in test `ResponseMessageTimeoutDownstreamDisconnect`.

So, in this PR I chosen the simplest solution with minimal changes to existing logic.

Reply:
FWIW, I am fixing some other issues(such as closeStream and byte info population) and will revisit this PR and here after those issues are fixed.

Reply:
Just recap here since it has been a bit while: 
We need the logging for the cases below:
1) onFinishProcessorCall. ---- normal response case
2) onRemoteClose  --- It is still valid to log onRemoteClose(only invalid to log after it is close). It includes: normal 
    remote close and remote close on error
3) onMessageTimeout --- we send our the message to ext_proc, but it timesout without response
4) onImmediateResponse ---  It is only needed because `onFinishProcessorCalls` 
    is called after `closeStream`. I think this logic could be tweaked but I think that change is a bit out of scope of this 
    PR. So I left a TODO for this case.

I also have the integration test with logging filter for all 4 cases above. 
PTAL, Thanks!